src/core/statistics/useStatistics.ts

```diff
@@ -30,39 +38,98 @@ export type TrackParameters = {
 
 export type EventAction = "send";
 
-export function useStatistics() {
+export const useCollectPageStatistics = () => {
+  const collectPageStatistics = ({ parameterName, trackedData }: EventData) => {
+    window._ti = window._ti || {};
+    window._ti[parameterName as string] = trackedData;
+  };
+
+  const resetAndCollectPageStatistics = ({
+    parameterName,
+    trackedData
+  }: EventData) => {
+    window._ti = {};
+    collectPageStatistics({ parameterName, trackedData });
+  };
+
+  return {
+    collectPageStatistics,
+    resetAndCollectPageStatistics
+  };
+};
+
+export function usePageStatistics() {
+  const config = useConfig();
+  const domain = config("mappDomainConfig");
+  const id = config("mappIdConfig");
+
+  const sendPageStatistics = ({ waitTime }: { waitTime: number }) => {
+    setTimeout(() => {
+      if (!domain || !id) {
+        // eslint-disable-next-line no-console
+        console.warn("⚠️ Mapp Domain or ID is not defined");
+        // This is to simulate the tracking request like the code in above for
+        // click events. Because domain and id are set as empty strings in Storybook
+        // The tracking script are not enabled. And therefore we console log the data
+        // eslint-disable-next-line no-console
+        console.log("Tracking: send, page", JSON.stringify(window._ti));
+        return;
+      }
+
+      const hasCollectedData = window._ti
+        ? Object.values(window._ti).some(
+            (val) => typeof val === "string" && val.trim() !== ""
+          )
+        : false;
```

Comment:
Consider if this can be replaced by [Lodash `isEmpty()`](https://lodash.com/docs/#isEmpty).